updates/build-log/memo/monitoring.md

```diff
@@ -0,0 +1,247 @@
+
+**Key Advantages:**
+
+* **Federated Queries**: Natively queries remote Parquet files and connects to our live **PostgreSQL** database in a single session.
+* **High Performance**: Columnar-vectorized query execution is highly optimized for analytical workloads.
+* **Zero-Dependency**: Runs entirely in-memory, simplifying our CI/CD environment setup.
+
+**PostgreSQL Integration:**
+The connection process involves loading the `postgres` extension and attaching our remote database as a read-only source.
+
+```typescript
+async function setupDuckDBConnections(): Promise<DuckDBConnection> {
+  const instance = await DuckDBInstance.create(':memory:');
+  const connection = await instance.connect();
+  
+  // Load extensions and attach remote database
+  await connection.runAndReadAll('LOAD postgres;');
+  await connection.runAndReadAll(`
+    ATTACH '${process.env.DB_CONNECTION_STRING}' AS memo_nft_db (TYPE postgres, READ_ONLY);
+  `);
+
+  return connection;
+}
+```
+
+### Security and compliance framework
+
+We designed the monitoring system with security and privacy as first-class citizens, ensuring all operations are secure and respectful of community data.
+
+#### Secrets management
+
+All credentials, API keys, and webhook URLs are managed as **GitHub Secrets**. They are securely injected into our workflows as environment variables at runtime and are never hardcoded in the repository.
+
+**Secure Environment Configuration:**
+
+```yaml
+env:
+  MEMO_NFT_DB_CONNECTION_STRING: ${{ secrets.MEMO_NFT_DB_CONNECTION_STRING }}
+  DISCORD_WEBHOOK_URL: ${{ secrets.DISCORD_WEBHOOK_URL_MEMO_LOGS }}
+  OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+```
+
+Our security posture is reinforced by:
+
+* **Principle of Least Privilege**: Secrets are scoped to only the workflows that require them.
+* **Read-Only Database Access**: The monitoring system connects to our database using a dedicated user with read-only permissions.
+* **Environment Isolation**: We use separate secrets and webhooks for development and production environments.
+
+#### Compliance and privacy
+
+We handle all data, especially community-related information, with strict adherence to privacy best practices. This involves data minimization and anonymization in all public-facing reports.
+
+**Address Truncation for Privacy:**
+To protect collector privacy, we truncate wallet addresses before they are included in any public notification.
+
+```typescript
+const topCollectors = collectorsResult.map((row) => ({
+  address: String(row.address || '').substring(0, 8) + '...', // Truncated for privacy
+  totalAmount: Number(row.total_amount)
+}));
+```
+
+### Operational management and performance
+
+We have established clear metrics and processes to ensure our monitoring system runs smoothly and delivers tangible value.
```

Comment:
Mấy đoạn kiểu này viết humble tý. dù gì nó cũng chỉ là website của nhà. đừng xài từ ngữ kiểu nó là sp make huge impact tới million of users...